File: pascal/strings/substring_modification.pas
(*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 *)

program SubstringModification;

uses strutils;

var
  cite: string = 'Stat rosa pristina nomine, nomina nuda tenemus.';
  modified_cite: string;

begin
  writeln('cite = ', cite);
  writeln();

  modified_cite := replacestr(cite, 'rosa', 'odonata');

  writeln('modified cite = ', modified_cite);
  writeln();

end.

File: c++/strings/strings/substring_modification.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>
#include <cstring>

int main()
{
  std::string cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
  std::cout << "cite = " << cite << std::endl << std::endl;

  std::string modified_cite = cite.replace(cite.find("rosa"), strlen("rosa"), "odonata");

  std::cout << "modified cite = " << modified_cite << std::endl << std::endl;
}
